tests/javascripts/support/helpers.js

```diff
@@ -0,0 +1,49 @@
+// helpers for mocking DOM interfaces
+// see https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model#DOM_interfaces
+
+// Base class for mocking an DOM API interfaces not in JSDOM
+class DOMInterfaceMock {
+
+  constructor (jest, spec) {
+
+    // set up methods so their calls can be tracked
+    // leave implementation/return values to the test
+    spec.methods.forEach(method => this[method] = jest.fn(() => {}));
+
+    // set up props
+    // any spies should be relative to the test so not set here
+    spec.props.forEach(prop => {
+
+      Object.defineProperty(this, prop, {
+        get: () => this[prop],
+        set: value => this[prop] = value
+      });
+
+    });
+
+  }
+
+}
+
+// Very basic class for stubbing the Range interface
+// Only contains methods required for current tests
+class RangeMock extends DOMInterfaceMock {
+
+  constructor (jest) {
+    super(jest, { props: [], methods: ['selectNodeContents', 'setStart'] });
```

Comment:
`setStart` isn't used anywhere, do we need it?

Reply:
No, it's used in version of this mock used in the admin app, which also has to support tests that alter the position of the range. Removed in https://github.com/alphagov/notifications-govuk-alerts/pull/105/commits/f1f3493e607847e8d7b738af605c811eacbe1415.